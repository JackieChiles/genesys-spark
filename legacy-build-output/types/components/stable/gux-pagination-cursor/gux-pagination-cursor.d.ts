import { JSX } from '../../../stencil-public-runtime';
export declare class GuxPaginationCursor {
  private i18n;
  private root;
  hasPrevious: boolean;
  hasNext: boolean;
  private guxPaginationCursorchange;
  private onButtonClick;
  componentWillLoad(): Promise<void>;
  render(): JSX.Element;
}
