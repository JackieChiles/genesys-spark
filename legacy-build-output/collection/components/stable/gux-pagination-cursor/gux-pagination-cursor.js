import { h } from '@stencil/core';
import { trackComponent } from '@utils/tracking/usage';
import { buildI18nForComponent } from '../../../i18n';
import translationResources from './i18n/en.json';
export class GuxPaginationCursor {
  constructor() {
    this.hasPrevious = false;
    this.hasNext = false;
  }
  onButtonClick(paginationDetail) {
    if ((paginationDetail === 'previous' && this.hasPrevious) ||
      (paginationDetail === 'next' && this.hasNext)) {
      this.guxPaginationCursorchange.emit(paginationDetail);
    }
  }
  async componentWillLoad() {
    trackComponent(this.root);
    this.i18n = await buildI18nForComponent(this.root, translationResources);
  }
  render() {
    return [
      h("gux-button-slot-beta", { accent: "secondary" }, h("button", { type: "button", title: this.i18n('previous'), disabled: !this.hasPrevious, onClick: () => this.onButtonClick('previous') }, h("gux-icon", { decorative: true, iconName: "chevron-small-left" }))),
      h("gux-button-slot-beta", { accent: "secondary" }, h("button", { type: "button", title: this.i18n('next'), disabled: !this.hasNext, onClick: () => this.onButtonClick('next') }, h("gux-icon", { decorative: true, iconName: "chevron-small-right" })))
    ];
  }
  static get is() { return "gux-pagination-cursor"; }
  static get encapsulation() { return "shadow"; }
  static get originalStyleUrls() {
    return {
      "$": ["gux-pagination-cursor.less"]
    };
  }
  static get styleUrls() {
    return {
      "$": ["gux-pagination-cursor.css"]
    };
  }
  static get properties() {
    return {
      "hasPrevious": {
        "type": "boolean",
        "mutable": false,
        "complexType": {
          "original": "boolean",
          "resolved": "boolean",
          "references": {}
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "has-previous",
        "reflect": false,
        "defaultValue": "false"
      },
      "hasNext": {
        "type": "boolean",
        "mutable": false,
        "complexType": {
          "original": "boolean",
          "resolved": "boolean",
          "references": {}
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "has-next",
        "reflect": false,
        "defaultValue": "false"
      }
    };
  }
  static get events() {
    return [{
        "method": "guxPaginationCursorchange",
        "name": "guxPaginationCursorchange",
        "bubbles": true,
        "cancelable": true,
        "composed": true,
        "docs": {
          "tags": [],
          "text": ""
        },
        "complexType": {
          "original": "GuxPaginationCursorDetail",
          "resolved": "\"next\" | \"previous\"",
          "references": {
            "GuxPaginationCursorDetail": {
              "location": "import",
              "path": "./gux-pagination-cursor.types"
            }
          }
        }
      }];
  }
  static get elementRef() { return "root"; }
}
