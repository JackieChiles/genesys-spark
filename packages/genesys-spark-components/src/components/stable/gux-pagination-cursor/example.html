<h1>gux-pagination-cursor</h1>
<h2>Default (no previous or next page available)</h2>
<div lang="en">
  <div id="default" class="table-placeholder"></div>
  <gux-pagination-cursor
    controls="default"
    label="Pagination for default example"
  ></gux-pagination-cursor>
</div>

<h2>Next page available</h2>
<div lang="en">
  <div id="next" class="table-placeholder"></div>
  <gux-pagination-cursor
    has-next
    controls="next"
    label="Pagination for next example"
  ></gux-pagination-cursor>
</div>

<h2>Next and previous page available</h2>
<div lang="en">
  <div id="next-previous" class="table-placeholder"></div>
  <gux-pagination-cursor
    has-previous
    has-next
    controls="next-previous"
    label="Pagination for next and previous example"
  ></gux-pagination-cursor>
</div>

<h2>Previous page available</h2>
<div lang="en">
  <div id="previous" class="table-placeholder"></div>
  <gux-pagination-cursor
    has-previous
    controls="previous"
    label="Pagination for previous example"
  ></gux-pagination-cursor>
</div>

<h2>Advanced Layout</h2>
<div lang="en">
  <div id="advanced" class="table-placeholder"></div>
  <gux-pagination-cursor
    layout="advanced"
    has-previous
    has-next
    controls="advanced"
    label="Pagination for advanced example"
  ></gux-pagination-cursor>
</div>

<h2>Advanced Layout with Item Per Page dropdown</h2>
<div lang="en">
  <div id="advanced-per-item" class="table-placeholder"></div>
  <gux-pagination-cursor
    layout="advanced"
    items-per-page="25"
    has-previous
    has-next
    controls="advanced-per-item"
    label="Pagination for advanced example with item per page dropdown"
  ></gux-pagination-cursor>
</div>

<style>
  .table-placeholder {
    padding: 10px;
    color: #f1f1f1;
    text-align: center;
    background-image: linear-gradient(
      45deg,
      #f1f1f1 25%,
      transparent 25%,
      transparent 50%,
      #f1f1f1 50%,
      #f1f1f1 75%,
      transparent 75%,
      #fff
    );
    background-size: 5px 5px;
  }

  .table-placeholder::before {
    padding: 5px;
    content: 'Table Placeholder';
    background-color: white;
  }
</style>
