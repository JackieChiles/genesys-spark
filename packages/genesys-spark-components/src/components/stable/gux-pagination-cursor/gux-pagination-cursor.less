@import (reference) '../../../style/color.less';
@import (reference) '../../../style/spacing.less';

:host {
  display: flex;
  flex-direction: row-reverse;
  flex-wrap: nowrap;
  align-content: stretch;
  align-items: flex-start;
  justify-content: space-between;
  padding: 12px;
  border-top: 1px solid @gux-grey-60;

  .gux-pagination-button-container {
    display: flex;
    flex-wrap: nowrap;
    justify-content: space-between;
  }

  button {
    color: @gux-black-50;

    &[disabled] {
      color: fade(@gux-black-50, 50%);
    }
  }

  .gux-button-align-content {
    display: flex;
    flex-direction: row;
    flex-wrap: nowrap;
    align-content: center;
    align-items: center;
    justify-content: center;

    > * {
      flex: 0 1 auto;
      align-self: auto;
      order: 0;
    }

    > .gux-icon-previous {
      padding-right: 4px;
    }

    > .gux-icon-next {
      padding-left: 4px;
    }
  }

  gux-icon {
    width: 16px;
    height: 16px;
  }

  gux-button-slot-beta {
    margin-right: 12px;
    margin-left: 12px;
  }
}
