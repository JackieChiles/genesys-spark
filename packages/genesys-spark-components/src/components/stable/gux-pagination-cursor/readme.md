# gux-pagination-cursor



<!-- Auto Generated Below -->


## Properties

| Property       | Attribute        | Description                                                                                    | Type                     | Default     |
| -------------- | ---------------- | ---------------------------------------------------------------------------------------------- | ------------------------ | ----------- |
| `hasNext`      | `has-next`       |                                                                                                | `boolean`                | `false`     |
| `hasPrevious`  | `has-previous`   |                                                                                                | `boolean`                | `false`     |
| `itemsPerPage` | `items-per-page` | Optional. Shows items per page dropdown when set. Only available with layout set to 'advanced' | `100 \| 25 \| 50 \| 75`  | `undefined` |
| `layout`       | `layout`         |                                                                                                | `"advanced" \| "simple"` | `'simple'`  |


## Events

| Event                       | Description | Type                                |
| --------------------------- | ----------- | ----------------------------------- |
| `guxitemsperpagechange`     |             | `CustomEvent<number>`               |
| `guxPaginationCursorchange` |             | `CustomEvent<"next" \| "previous">` |


## Dependencies

### Depends on

- [gux-button-slot](../gux-button-slot)
- [gux-icon](../gux-icon)

### Graph
```mermaid
graph TD;
  gux-pagination-cursor --> gux-button-slot
  gux-pagination-cursor --> gux-icon
  style gux-pagination-cursor fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
