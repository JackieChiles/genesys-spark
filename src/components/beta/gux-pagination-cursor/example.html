<h1>Default (no previous or next page available)</h1>
<div lang="en">
  <div class="table-placeholder"></div>
  <gux-pagination-cursor-beta></gux-pagination-cursor-beta>
</div>

<h1>Next page available</h1>
<div lang="en">
  <div class="table-placeholder"></div>
  <gux-pagination-cursor-beta has-next></gux-pagination-cursor-beta>
</div>

<h1>Next and previous page available</h1>
<div lang="en">
  <div class="table-placeholder"></div>
  <gux-pagination-cursor-beta
    has-previous
    has-next
  ></gux-pagination-cursor-beta>
</div>

<h1>Previous page available</h1>
<div lang="en">
  <div class="table-placeholder"></div>
  <gux-pagination-cursor-beta has-previous></gux-pagination-cursor-beta>
</div>

<style>
  .table-placeholder {
    padding: 10px;
    color: #f1f1f1;
    text-align: center;
    background-image: linear-gradient(
      45deg,
      #f1f1f1 25%,
      transparent 25%,
      transparent 50%,
      #f1f1f1 50%,
      #f1f1f1 75%,
      transparent 75%,
      #fff
    );
    background-size: 5px 5px;
  }

  .table-placeholder::before {
    padding: 5px;
    content: 'Table Placeholder';
    background-color: white;
  }
</style>
