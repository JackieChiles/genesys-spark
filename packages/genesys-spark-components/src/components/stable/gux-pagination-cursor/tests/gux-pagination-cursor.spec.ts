import { newSpecPage } from '@test/specTestUtils';
import { GuxPaginationCursor } from '../gux-pagination-cursor';

const components = [GuxPaginationCursor];
const language = 'en';

describe('gux-pagination-cursor', () => {
  describe('#render', () => {
    [
      '<gux-pagination-cursor controls="id1" label="Example label"></gux-pagination-cursor>',
      '<gux-pagination-cursor has-next controls="id2" label="Example label"></gux-pagination-cursor>',
      '<gux-pagination-cursor has-previous has-next controls="id3" label="Example label"></gux-pagination-cursor>',
      '<gux-pagination-cursor has-previous controls="id4" label="Example label"></gux-pagination-cursor>',
      '<gux-pagination-cursor layout="advanced" controls="id5" label="Example label"></gux-pagination-cursor>'
    ].forEach((html, index) => {
      it(`should render as expected (${index + 1})`, async () => {
        const page = await newSpecPage({ components, html, language });

        expect(page.rootInstance).toBeInstanceOf(GuxPaginationCursor);
        expect(page.root).toMatchSnapshot();
      });
    });
  });

  describe('guxPaginationCursorchange', () => {
    it('should fire guxPaginationCursorchange(previous) event when enabled previous button is clicked', async () => {
      const html =
        '<gux-pagination-cursor has-previous controls="previous" label="Example label"></gux-pagination-cursor>';
      const page = await newSpecPage({ components, html, language });
      const element = page.root as HTMLElement;
      const [previousButton] = Array.from(
        element.shadowRoot.querySelectorAll('button')
      ) as HTMLElement[];
      const guxPaginationCursorchangeSpy = jest.fn();

      page.win.addEventListener(
        'guxPaginationCursorchange',
        guxPaginationCursorchangeSpy
      );

      previousButton.click();
      await page.waitForChanges();

      expect(guxPaginationCursorchangeSpy).toHaveBeenCalledWith(
        expect.objectContaining({ detail: 'previous' })
      );
    });

    it('should fire guxPaginationCursorchange(next) event when enabled next button is clicked', async () => {
      const html =
        '<gux-pagination-cursor has-next controls="next" label="Example label"></gux-pagination-cursor>';
      const page = await newSpecPage({ components, html, language });
      const element = page.root as HTMLElement;
      const [, nextButton] = Array.from(
        element.shadowRoot.querySelectorAll('button')
      ) as HTMLElement[];
      const guxPaginationCursorchangeSpy = jest.fn();

      page.win.addEventListener(
        'guxPaginationCursorchange',
        guxPaginationCursorchangeSpy
      );

      nextButton.click();
      await page.waitForChanges();

      expect(guxPaginationCursorchangeSpy).toHaveBeenCalledWith(
        expect.objectContaining({ detail: 'next' })
      );
    });

    it('should not fire guxPaginationCursorchange(previous) event when disabled previous button is clicked', async () => {
      const html =
        '<gux-pagination-cursor controls="previous-disabled" label="Example label"></gux-pagination-cursor>';
      const page = await newSpecPage({ components, html, language });
      const element = page.root as HTMLElement;
      const [previousButton] = Array.from(
        element.shadowRoot.querySelectorAll('button')
      ) as HTMLElement[];
      const guxPaginationCursorchangeSpy = jest.fn();

      page.win.addEventListener(
        'guxPaginationCursorchange',
        guxPaginationCursorchangeSpy
      );

      previousButton.click();
      await page.waitForChanges();

      expect(guxPaginationCursorchangeSpy).not.toHaveBeenCalled();
    });

    it('should not fire guxPaginationCursorchange(next) event when disabled next button is clicked', async () => {
      const html =
        '<gux-pagination-cursor controls="next-disabled" label="Example label"></gux-pagination-cursor>';
      const page = await newSpecPage({ components, html, language });
      const element = page.root as HTMLElement;
      const [, nextButton] = Array.from(
        element.shadowRoot.querySelectorAll('button')
      ) as HTMLElement[];
      const guxPaginationCursorchangeSpy = jest.fn();

      page.win.addEventListener(
        'guxPaginationCursorchange',
        guxPaginationCursorchangeSpy
      );

      nextButton.click();
      await page.waitForChanges();

      expect(guxPaginationCursorchangeSpy).not.toHaveBeenCalled();
    });
  });
});
