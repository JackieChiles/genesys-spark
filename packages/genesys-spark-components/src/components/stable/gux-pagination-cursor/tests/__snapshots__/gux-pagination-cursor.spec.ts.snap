// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-pagination-cursor #render should render as expected (1) 1`] = `
<gux-pagination-cursor controls="id1" label="Example label">
  <mock:shadow-root>
    <nav aria-label="Example label" class="gux-pagination-button-container">
      <gux-button-slot accent="ghost">
        <button aria-controls="id1" class="gux-simple-button" disabled="" type="button">
          <gux-icon iconname="custom/chevron-left-small-regular" screenreader-text="Previous"></gux-icon>
        </button>
      </gux-button-slot>
      <gux-button-slot accent="ghost">
        <button aria-controls="id1" class="gux-simple-button" disabled="" type="button">
          <gux-icon iconname="custom/chevron-right-small-regular" screenreader-text="Next"></gux-icon>
        </button>
      </gux-button-slot>
    </nav>
  </mock:shadow-root>
</gux-pagination-cursor>
`;

exports[`gux-pagination-cursor #render should render as expected (2) 1`] = `
<gux-pagination-cursor controls="id2" has-next="" label="Example label">
  <mock:shadow-root>
    <nav aria-label="Example label" class="gux-pagination-button-container">
      <gux-button-slot accent="ghost">
        <button aria-controls="id2" class="gux-simple-button" disabled="" type="button">
          <gux-icon iconname="custom/chevron-left-small-regular" screenreader-text="Previous"></gux-icon>
        </button>
      </gux-button-slot>
      <gux-button-slot accent="ghost">
        <button aria-controls="id2" class="gux-simple-button" type="button">
          <gux-icon iconname="custom/chevron-right-small-regular" screenreader-text="Next"></gux-icon>
        </button>
      </gux-button-slot>
    </nav>
  </mock:shadow-root>
</gux-pagination-cursor>
`;

exports[`gux-pagination-cursor #render should render as expected (3) 1`] = `
<gux-pagination-cursor controls="id3" has-next="" has-previous="" label="Example label">
  <mock:shadow-root>
    <nav aria-label="Example label" class="gux-pagination-button-container">
      <gux-button-slot accent="ghost">
        <button aria-controls="id3" class="gux-simple-button" type="button">
          <gux-icon iconname="custom/chevron-left-small-regular" screenreader-text="Previous"></gux-icon>
        </button>
      </gux-button-slot>
      <gux-button-slot accent="ghost">
        <button aria-controls="id3" class="gux-simple-button" type="button">
          <gux-icon iconname="custom/chevron-right-small-regular" screenreader-text="Next"></gux-icon>
        </button>
      </gux-button-slot>
    </nav>
  </mock:shadow-root>
</gux-pagination-cursor>
`;

exports[`gux-pagination-cursor #render should render as expected (4) 1`] = `
<gux-pagination-cursor controls="id4" has-previous="" label="Example label">
  <mock:shadow-root>
    <nav aria-label="Example label" class="gux-pagination-button-container">
      <gux-button-slot accent="ghost">
        <button aria-controls="id4" class="gux-simple-button" type="button">
          <gux-icon iconname="custom/chevron-left-small-regular" screenreader-text="Previous"></gux-icon>
        </button>
      </gux-button-slot>
      <gux-button-slot accent="ghost">
        <button aria-controls="id4" class="gux-simple-button" disabled="" type="button">
          <gux-icon iconname="custom/chevron-right-small-regular" screenreader-text="Next"></gux-icon>
        </button>
      </gux-button-slot>
    </nav>
  </mock:shadow-root>
</gux-pagination-cursor>
`;

exports[`gux-pagination-cursor #render should render as expected (5) 1`] = `
<gux-pagination-cursor controls="id5" label="Example label" layout="advanced">
  <mock:shadow-root>
    <nav aria-label="Example label" class="gux-pagination-button-container">
      <gux-button-slot accent="ghost">
        <button disabled="" type="button">
          <div class="gux-button-align-content">
            <gux-icon decorative="" iconname="custom/chevron-left-small-regular"></gux-icon>
            <span>
              Previous
            </span>
          </div>
        </button>
      </gux-button-slot>
      <gux-button-slot accent="ghost">
        <button disabled="" type="button">
          <div class="gux-button-align-content">
            <span>
              Next
            </span>
            <gux-icon decorative="" iconname="custom/chevron-right-small-regular"></gux-icon>
          </div>
        </button>
      </gux-button-slot>
    </nav>
  </mock:shadow-root>
</gux-pagination-cursor>
`;
